python/tests/test_run_finished_always_emitted.py

```
@@ -0,0 +1,184 @@
+
+    def __init__(self) -> None:
+        super().__init__(endpoint="http://localhost", api_key="test")
+        self.received: List[ScenarioEvent] = []
+
+    async def post_event(self, event: ScenarioEvent) -> Dict[str, Any]:
+        self.received.append(event)
+        return {}
+
+
+class _Agent(AgentAdapter):
+    async def call(self, input: AgentInput) -> AgentReturnTypes:
```

Ruff (0.16.1)

[error] Function argument `input` is shadowing a Python builtin

(A002)